### PR TITLE
feat: added ngram algorithm

### PR DESCRIPTION
**Pull request by 9harshit**
### Describe your change:

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Please rebase on the latest master.
1. `git checkout master`
2. `git pull <remote> master`
3. `git rebase master <local branch name>`

File: strings/ngram.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    :param ngram_size: int
+    :return: list
+
+    >>> create_ngram("I am a sentence", 2)
+    ['I ', ' a', 'am', 'm ', ' a', 'a ', ' s', 'se', 'en', 'nt', 'te', 'en', 'nc', 'ce']
+    >>> create_ngram("I am an NLPer", 2)
+    ['I ', ' a', 'am', 'm ', ' a', 'an', 'n ', ' N', 'NL', 'LP', 'Pe', 'er']
+    """
+
+    ngrams = []
```
Comment (cclauss) [inline strings/ngram.py:21]: Could this be a list comprehension?

File: arithmetic_analysis/in_static_equilibrium.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ def polar_force(
     Resolves force along rectangular components.
     (force, angle) => (force_x, force_y)
     >>> polar_force(10, 45)
-    [7.071067811865477, 7.0710678118654755]
+    [7.0710678118654755, 7.071067811865475]
```
Comment (cclauss) [inline arithmetic_analysis/in_static_equilibrium.py:16]: Do we need to use https://docs.python.org/3/library/math.html#math.isclose to make this test pass on multiple cpu architectures?

**Review by cclauss — DISMISSED** [review]

File: arithmetic_analysis/in_static_equilibrium.py (hunk 2 above)
Comment (dhruvmanila) [reply arithmetic_analysis/in_static_equilibrium.py:16]: We could use `round` instead.

File: strings/ngram.py (hunk 1 above)
Comment (9harshit) [reply strings/ngram.py]: List comprehension added

File: strings/ngram.py [hunk 3]
```diff
@@ -18,9 +18,9 @@ def create_ngram(sentence: str, ngram_size: int) -> list:
     ['I ', ' a', 'am', 'm ', ' a', 'an', 'n ', ' N', 'NL', 'LP', 'Pe', 'er']
     """
 
-    ngrams = []
-    for i in range(len(sentence) - ngram_size + 1):
-        ngrams.append(sentence[i : i + ngram_size])
+    ngrams = [
+        sentence[i : i + ngram_size] for i in range(len(sentence) - ngram_size + 1)
+    ]
     return ngrams
```
Comment (cclauss) [inline strings/ngram.py:24]:
```suggestion
    return [sentence[i : i + ngram_size] for i in range(len(sentence) - ngram_size + 1)]
```

File: strings/ngram.py [hunk 4]
```diff
@@ -18,9 +18,9 @@ def create_ngram(sentence: str, ngram_size: int) -> list:
     ['I ', ' a', 'am', 'm ', ' a', 'an', 'n ', ' N', 'NL', 'LP', 'Pe', 'er']
     """
```
Comment (cclauss) [inline strings/ngram.py:19]: Please add a test...
```suggestion
    >>> create_ngram("This is short", 50)
    ???
    """
```

Comment (cclauss) [reply strings/ngram.py:21]: Please do not mark this resolved until after the test passes.

Please replace the `???` with `[]`.  Is `[]` the list of ngrams that we expected?

File: strings/ngram.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+"""
+
+
+def create_ngram(sentence: str, ngram_size: int) -> list:
+    """
+    Create ngrams from a sentence
+    :param sentence: str
+    :param ngram_size: int
+    :return: list
```
Comment (cclauss) [inline strings/ngram.py:13]: If these lines only repeat information that has already been stated in the type hints then they are not needed.
```suggestion
```

File: strings/ngram.py (hunk 4 above)
Comment (9harshit) [reply strings/ngram.py:18]: Yes, changes done.

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**